bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromHandler.java

```diff
@@ -0,0 +1,169 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MyStromHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Frank - Initial contribution
+ */
+@NonNullByDefault
+public class MyStromHandler extends BaseThingHandler {
+
+    private static class MyStromReport {
+
+        public float power;
+        public boolean relay;
+        public float temperature;
+    }
+
+    private static final int HTTP_OK_CODE = 200;
+    private static final String COMMUNICATION_ERROR = "Error while communicating to myStrom: ";
+
+    private final Logger logger = LoggerFactory.getLogger(MyStromHandler.class);
+
+    private @Nullable MyStromConfiguration config;
+    private HttpClient httpClient;
+    private String hostname = "";
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private ExpiringCache<MyStromReport> cache = new ExpiringCache<>(Duration.ofSeconds(3), this::getReport);
+    private final Gson gson = new Gson();
+
+    public MyStromHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                pollDevice();
+            } else {
+                if (command instanceof OnOffType && CHANNEL_SWITCH.equals(channelUID.getId())) {
+                    sendHttpGet("relay?state=" + (command == OnOffType.ON ? "1" : "0"));
+                    scheduler.schedule(this::pollDevice, 500, TimeUnit.MILLISECONDS);
+                }
+
+            }
+        } catch (MyStromException e) {
+            logger.warn(COMMUNICATION_ERROR, e);
+        }
+    }
+
+    private @Nullable MyStromReport getReport() {
+        try {
+            String returnContent = sendHttpGet("report");
+            MyStromReport report = gson.fromJson(returnContent, MyStromReport.class);
+            updateStatus(ThingStatus.ONLINE);
+            return report;
+        } catch (MyStromException e) {
+            logger.debug(COMMUNICATION_ERROR, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, COMMUNICATION_ERROR);
+            return null;
+        }
+    }
+
+    private void pollDevice() {
+        MyStromReport report = cache.getValue();
+        if (report != null) {
+            updateState(CHANNEL_SWITCH, report.relay ? OnOffType.ON : OnOffType.OFF);
+            updateState(CHANNEL_POWER, QuantityType.valueOf(report.power, WATT));
+            updateState(CHANNEL_TEMPERATURE, QuantityType.valueOf(report.temperature, CELSIUS));
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, COMMUNICATION_ERROR);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
removed the log entry